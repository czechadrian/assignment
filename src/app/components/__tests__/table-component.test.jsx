import React from "react";
import { useDispatch, useSelector } from "react-redux";
import "jest-styled-components";
import renderer from "react-test-renderer";
import TableComponent from "../table-component";
import {
  removeSelectedCommentAction,
  selectCommentAction,
} from "../../screens/actions/selected-comments-actions";

const payload = require("../../../api-wrapper/test-helper/comments.data");
jest.mock("../../screens/actions/selected-comments-actions", () => {
  return {
    selectCommentAction: jest.fn(),
    removeSelectedCommentAction: jest.fn(),
  };
});

jest.mock("react-redux", () => {
  return {
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  };
});

describe("TableComponent", () => {
  const dispathMock = jest.fn();
  beforeEach(() => {
    useSelector.mockImplementation((selectorFn) => selectorFn());
    useDispatch.mockReturnValue(dispathMock);

    selectCommentAction.mockReturnValue(() => jest.fn());
    removeSelectedCommentAction.mockReturnValue(() => jest.fn());

    jest.clearAllMocks();
  });
  test("selected comments component test", () => {
    const tree = renderer
      .create(
        <TableComponent
          rows={payload.payload.comments}
          homePageTable={false}
          action={removeSelectedCommentAction}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("homepage component test", () => {
    const tree = renderer
      .create(
        <TableComponent
          rows={payload.payload.comments}
          homePageTable={true}
          action={selectCommentAction}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
