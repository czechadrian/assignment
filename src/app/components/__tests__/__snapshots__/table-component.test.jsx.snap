// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent homepage component test 1`] = `
.c2 {
  align: right;
}

.c1 {
  aria-label: "simple table";
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10% 0;
}

<main
  className="c0"
>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      className="MuiTable-root c1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head c2"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head c2"
            scope="col"
          >
            E-mail
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head c2"
            scope="col"
          >
            First 20 characters
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head c2"
            scope="col"
          >
            Add/Remove to Selected
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            id labore ex et quam laborum
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            Eliseo@gardner.biz
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            laudantium enim quas
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Added
              </span>
            </button>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            quo vero reiciendis velit similique earum
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            Jayne_Kuhic@sydney.com
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            est natus enim nihil
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Add
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</main>
`;

exports[`TableComponent selected comments component test 1`] = `
.c2 {
  align: right;
}

.c1 {
  aria-label: "simple table";
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 10% 0;
}

<main
  className="c0"
>
  <div
    className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <table
      className="MuiTable-root c1"
      role={null}
    >
      <thead
        className="MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head c2"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head c2"
            scope="col"
          >
            E-mail
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head c2"
            scope="col"
          >
            First 20 characters
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head c2"
            scope="col"
          >
            Add/Remove to Selected
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            id labore ex et quam laborum
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            Eliseo@gardner.biz
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            laudantium enim quas
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Remove
              </span>
            </button>
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body"
            role="cell"
            scope="row"
          >
            quo vero reiciendis velit similique earum
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            Jayne_Kuhic@sydney.com
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            est natus enim nihil
          </td>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body c2"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Removed
              </span>
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</main>
`;
